app/models/ranking/popular_posts.rb

```diff
@@ -0,0 +1,54 @@
+class Ranking::PopularPosts
+  def ranked_posts(site)
+    @site = site
```

Comment:
Do not set instance vars except constructor.
It is difficult to find what happen in the object (instance of `Ranking::PopularPosts`) if it has state.
If you don't have any reason, please don't change state.